diff -u emacs-jabber/jabber-iq.el.orig emacs-jabber/jabber-iq.el
--- emacs-jabber/jabber-iq.el.orig	2007-07-12 22:45:54 +0800
+++ emacs-jabber/jabber-iq.el	2007-07-12 23:06:51 +0800
@@ -43,6 +43,18 @@
   :group 'jabber
   :type 'hook)
 
+(defgroup jabber-browse nil "browse display options"
+  :group 'jabber)
+
+(defcustom jabber-browse-buffer-format "*-jabber-browse:-%n-*"
+  "The format specification for the name of browse buffers.
+
+These fields are available at this moment:
+
+%n   JID to browse"
+  :type 'string
+  :group 'jabber-browse)
+
 (defun jabber-browse-mode ()
 "\\{jabber-browse-mode-map}"
   (kill-all-local-variables)
@@ -152,7 +164,9 @@
   (let ((from (or (jabber-xml-get-attribute xml-data 'from) jabber-server))
 	(xmlns (jabber-iq-xmlns xml-data))
 	(type (jabber-xml-get-attribute xml-data 'type)))
-    (with-current-buffer (get-buffer-create (concat "*-jabber-browse-:-" from "-*"))
+    (with-current-buffer (get-buffer-create (format-spec jabber-browse-buffer-format
+                                                         (list
+                                                          (cons ?n from))))
       (if (not (eq major-mode 'jabber-browse-mode))
 	  (jabber-browse-mode))
